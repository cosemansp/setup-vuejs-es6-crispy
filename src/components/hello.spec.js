import { mount, shallow, mountConfig } from 'test/helper';
import Hello from './hello';

describe('Hello Component', () => {
  it('should have the data hook', () => {
    expect(typeof Hello.data).toBe('function');
  });

  it('should have a default "msg" set to "Welcome"', () => {
    const defaultData = Hello.data();
    expect(defaultData.msg).toMatch('Welcome');
  });

  it('should renders correctly', () => {
    // render component as a single unit
    const fixture = shallow(Hello);
    fixture.setData({ msg: 'Hello World' });
    expect(fixture).toMatchSnapshot();
  });

  it('should renders correctly with childs', () => {
    // render component with children
    const fixture = mount(Hello);
    expect(fixture).toMatchSnapshot();
  });

  it('should renders config', () => {
    const componentConfig = {
      template: `
        <div>
          <hello ref="component"></hello>
        </div>
      `,
      components: {
        Hello,
      },
    };
    const fixture = mountConfig(componentConfig);
    expect(fixture).toMatchSnapshot();
    expect(fixture.vm.$refs.component).toBeTruthy();
    expect(fixture.vm.$refs.component.msg).toMatch('Welcome');
  });
});

