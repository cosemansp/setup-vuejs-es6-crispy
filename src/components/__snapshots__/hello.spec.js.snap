// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component should renders config 1`] = `

<div>
  <div class="hello">
    <h1>
      Welcome to Your Vue.js App
    </h1>
    <div>
      <h1>
        Hello
      </h1>
    </div>
  </div>
</div>

`;

exports[`Hello Component should renders correctly 1`] = `

<div class="hello">
  <h1>
    Hello World
  </h1>
  <sample>
  </sample>
</div>

`;

exports[`Hello Component should renders correctly with childs 1`] = `

<div class="hello">
  <h1>
    Welcome to Your Vue.js App
  </h1>
  <div>
    <h1>
      Hello
    </h1>
  </div>
</div>

`;
